app/javascript/hyacinth_v1/components/digital_objects/metadata/MetadataForm.js

```diff
@@ -78,78 +125,75 @@ function MetadataForm(props) {
     return newObject;
   };
 
-  const keepEnabledFields = (enabledFieldIds, children) => {
-    return children.map((c) => {
-      switch (c.type) {
-        case 'DynamicFieldGroup':
-          c.children = keepEnabledFields(enabledFieldIds, c.children);
-          return c.children.length > 0 ? c : null;
-        case 'DynamicField':
-          return enabledFieldIds.includes(c.id) ? c : null;
-        default:
-          return c;
-      }
-    }).filter(c => c !== null);
-  };
-
-  const renderCategory = (category) => {
-    const { displayLabel, children } = category;
-    return (
-      <div key={displayLabel}>
-        <h4 className="text-orange">{displayLabel}</h4>
-        {
-          children.map(fieldGroup => (
-            <FieldGroupArray
-              key={`array_${fieldGroup.stringKey}`}
-              dynamicFieldGroup={fieldGroup}
-              value={descriptiveMetadata[fieldGroup.stringKey]}
-              defaultValue={defaultFieldData[fieldGroup.stringKey][0]}
-              onChange={v => onChange(fieldGroup.stringKey, v)}
-            />
-          ))
-        }
-      </div>
-    );
-  };
+  const keepEnabledFields = (enabledFieldIds, children) => children.map((c) => {
+    switch (c.type) {
+      case 'DynamicFieldGroup':
+        c.children = keepEnabledFields(enabledFieldIds, c.children);
+        return c.children.length > 0 ? c : null;
+      case 'DynamicField':
+        if (enabledFieldIds.includes(c.id)) return c;
+        // if the ids are fetched in queries ID scalars and JSON, strings must be compared
+        if (enabledFieldIds.includes(String(c.id))) return c;
+        return null;
+      default:
+        return c;
+    }
+  }).filter(c => c !== null);
 
   // TODO: Replace effect below with GraphQL when we have a GraphQL DynamicFieldCategories API
```

Comment:
The referenced effect is gone!  Okay to delete comment?

Reply:
```suggestion
```